code/game/objects/structures/lattice.dm

```diff
@@ -8,33 +8,27 @@
 	w_class = ITEM_SIZE_NORMAL
 	layer = LATTICE_LAYER
 	color = COLOR_STEEL
-	var/init_material = MAT_STEEL
+	material = MAT_STEEL
 	obj_flags = OBJ_FLAG_NOFALL
+	material_alteration = MAT_FLAG_ALTERATION_ALL
 
-/obj/structure/lattice/get_material()
-	return material
-
-/obj/structure/lattice/Initialize(mapload, var/new_material)
+/obj/structure/lattice/Initialize()
 	. = ..()
-	if(!(istype(src.loc, /turf/space) || istype(src.loc, /turf/simulated/open)))
-		return INITIALIZE_HINT_QDEL
-	if(!new_material)
-		new_material = init_material
-	material = SSmaterials.get_material_datum(new_material)
 	if(!istype(material))
 		return INITIALIZE_HINT_QDEL
-
-	SetName("[material.display_name] lattice")
-	desc = "A lightweight support [material.display_name] lattice."
-	color =  material.icon_colour
-
+	if(!istype(src.loc, /turf/space) && !istype(src.loc, /turf/simulated/open))
+		return INITIALIZE_HINT_QDEL
 	for(var/obj/structure/lattice/LAT in loc)
 		if(LAT != src)
 			crash_with("Found multiple lattices at '[log_info_line(loc)]'")
 			qdel(LAT)
-	update_icon()
-	if(!mapload)
-		update_neighbors()
+	update_neighbors()
```

Comment:
I believe these calls are generally not safe until `LateInitialize`, as they can make further update calls (icons, notably) which require `Initialize` to have run to not runtime (possibly due to materials setup). That's roughly why you see all the LateInitialize everywhere; it used to do New/Initialize abuse or spawns to deal with the same issue.

Reply:
Implemented LateInitialize, not sure why this isn't marked as outdated.